src/parser/waypoint.rs

```diff
@@ -75,54 +76,63 @@ pub fn consume<R: Read>(reader: &mut Peekable<Events<R>>) -> Result<Waypoint> {
                     }
                     "ele" => {
                         // Cast the elevation to an f64, from a string.
-                        elevation = Some(string::consume(reader)?
+                        elevation = Some(string::consume(reader, version)?
                             .parse()
                             .chain_err(|| "error while casting elevation to f64")?)
                     }
-                    "time" => time = Some(time::consume(reader)?),
-                    "name" => wptname = Some(string::consume(reader)?),
-                    "cmt" => comment = Some(string::consume(reader)?),
-                    "desc" => description = Some(string::consume(reader)?),
-                    "src" => source = Some(string::consume(reader)?),
-                    "link" => links.push(link::consume(reader)?),
-                    "sym" => symbol = Some(string::consume(reader)?),
-                    "type" => _type = Some(string::consume(reader)?),
+                    "speed" => {
+                        if version == GpxVersion::Gpx10 {
+                            // Speed is from GPX 1.0
+                            // Ignore.
```

Comment:
Might this be something we should add to parse-able data, noting it's only for GPX 1.0?

Reply:
I added it in Waypoint now